scripts/jenkins/Jenkinsfile

```diff
@@ -0,0 +1,538 @@
+// FIXME sloc publish errors
+// XXX add missing jobs
+// TODO have a per plugin/binding deps in Dockerfile for easier maintanence
+// TODO add warnings plugins to scan for compiler warnings
+
+// # Imports #
+import java.text.SimpleDateFormat
+
+// # Set Variables #
+CMPVERSION = '0.8.22'
+VERSION = '0.8.23'
+
+docker_node_label = 'docker'
+docker_image_prefix = 'build-elektra-'
+
+// ## Setup log discarder ##
+properties([
+    buildDiscarder(logRotator(numToKeepStr: '60', artifactNumToKeepStr: '60'))
+])
+
+// ## Define Docker images globals ##
+DOCKER_IMAGES = [
+    sid: [
+        id: docker_image_prefix + 'debian-sid',
+        context: "./scripts/docker/debian/sid",
+        file: "./scripts/docker/debian/sid/Dockerfile"
+    ],
+    stretch: [
+        id: docker_image_prefix + 'debian-stretch',
+        context: "./scripts/docker/debian/stretch",
+        file: "./scripts/docker/debian/stretch/Dockerfile",
+    ],
+    stretch_minimal: [
+        id: docker_image_prefix + 'debian-stretch-minimal',
+        context: "./scripts/docker/debian/stretch",
+        file: "./scripts/docker/debian/stretch/Dockerfile.minimal",
+    ],
+    stretch_doc: [
+        id: docker_image_prefix + 'debian-stretch-doc',
+        context: "./scripts/docker/debian/stretch",
+        file: "./scripts/docker/debian/stretch/Dockerfile.doc",
+    ],
+    jessie: [
+        id: docker_image_prefix + 'debian-jessie',
+        context: "./scripts/docker/debian/jessie",
+        file: "./scripts/docker/debian/jessie/Dockerfile",
+    ],
+    xenial: [
+        id: docker_image_prefix + 'ubuntu-xenial',
+        context: "./scripts/docker/ubuntu/xenial",
+        file: "./scripts/docker/ubuntu/xenial/Dockerfile",
+    ]
+]
+
+// ## Define Build Flag globals ##
+BUILD_FLAGS_STANDARD = [
+    'SITE': '${STAGE_NAME}',
+    'KDB_DB_SYSTEM': '${HOME}/.config/kdb/system',
+    'KDB_DB_SPEC': '${HOME}/.config/kdb/spec',
+    'CMAKE_INSTALL_PREFIX': '${WORKSPACE}/system',
+    'INSTALL_SYSTEM_FILES': 'OFF',
+    'BUILD_DOCUMENTATION': 'OFF'
+]
+
+BUILD_FLAGS_ALL = [
+    'BINDINGS': 'ALL;EXPERIMENTAL;-DEPRECATED',
+    'PLUGINS': 'ALL;EXPERIMENTAL;-DEPRECATED',
+    'TOOLS': 'ALL'
+]
+
+BUILD_FLAGS_COVERAGE = [
+    'ENABLE_COVERAGE': 'ON'
+]
+
+BUILD_FLAGS_JAVA = [
+    'JAVA_AWT_INCLUDE_PATH': '/usr/lib/jvm/java-8-openjdk-amd64/include/'
+]
+
+BUILD_FLAGS_CLANG = [
+    'CMAKE_C_COMPILER': 'clang',
+    'CMAKE_CXX_COMPILER': 'clang++'
+]
+
+BUILD_FLAGS_ASAN = ['ENABLE_ASAN': 'ON']
+BUILD_FLAGS_DEBUG = ['ENABLE_DEBUG': 'ON']
+BUILD_FLAGS_LOGGER = ['ENABLE_LOGGER': 'ON']
+
+// ## Define Test Flag globals ##
+TEST_MEM = 'mem'
+TEST_NOKDB = 'nokdb'
+TEST_ALL = 'all'
+TEST_INSTALL = 'install'
+
+// ## DATE ##
+NOW = new Date()
+
+// # Determine what needs to be build #
+target = ".*"
```

Comment:
Does this mean by default everything is build? (The comment directly above is confusing)

Reply:
It plans to build everything per default. But if some of the fast builds fail it will not queue the more intensive ones which occur later